src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/MessageList.vue

```diff
@@ -1,43 +1,72 @@
-<!-- eslint-disable vue/no-v-html -->
 <template>
   <v-list lines="two" class="flex-grow-1 message-list">
-    <MessageListItem v-for="(message, index) in messageStore.paginatedMessages" :key="index" :message="message" />
-    <v-pagination
-      v-if="messageStore.totalPages > 1"
-      v-model="currentPage"
-      size="small"
-      class="mt-4"
-      elevation="2"
-      :length="messageStore.totalPages"
-    ></v-pagination>
+    <MessageListItem v-for="(message, index) in messages" :key="index" :message="message" @message-read="fetchMessages(page)" />
+    <v-pagination v-if="messageCount > 1" v-model="currentPage" size="small" class="mt-4" elevation="2" :length="totalPages"></v-pagination>
   </v-list>
 </template>
 
 <script lang="ts">
 import { defineComponent } from "vue";
 
+import { getMessages } from "@/api/message";
 import MessageListItem from "@/components/MessageListItem.vue";
 import { useMessageStore } from "@/store/message";
+import type { Components } from "@/types/openapi";
+
+const PAGE_SIZE = 10;
 
 export default defineComponent({
   name: "MessageList",
   components: { MessageListItem },
   setup() {
     const messageStore = useMessageStore();
+
     return { messageStore };
   },
+  data() {
+    return {
+      messages: [] as Components.Schemas.Communication[],
+      messageCount: 0,
+      page: 1,
+    };
+  },
   computed: {
+    totalPages() {
+      return Math.ceil(this.messageCount / PAGE_SIZE);
+    },
     currentPage: {
       get() {
-        return this.messageStore.currentPage;
+        return this.page;
       },
-      set(value: number) {
-        this.messageStore.setPage(value);
+      set(newValue: number) {
+        this.page = newValue;
+        this.resetMessageStore();
+        this.fetchMessages(newValue);
+        window.scrollTo({
+          top: 0,
+          behavior: "smooth",
+        });
       },
     },
   },
-  created() {
-    this.messageStore.fetchMessages();
+  mounted() {
+    this.resetMessageStore();
+    this.fetchMessages(this.page);
+  },
+  methods: {
+    async fetchMessages(page: number) {
+      const params = {
+        page,
+        pageSize: PAGE_SIZE,
+      };
+      const response = await getMessages(params);
+      this.messages = response.data?.communications || [];
+      this.messageCount = response.data?.totalMessagesCount || 0;
+    },
+    async resetMessageStore() {
+      this.messageStore.currentMessage = null;
```

Comment:
If you are reseting the whole store you can just use `this.messageStore.$reset()` and skip this method

Reply:
Done..